tests/components/influxdb/test_sensor.py

```diff
@@ -0,0 +1,453 @@
+                "port": "9000",
+                "path": "path",
+                "username": "user",
+                "password": "pass",
+                "database": "db",
+                "verify_ssl": "true",
+                "queries": [
+                    {
+                        "name": "test",
+                        "unit_of_measurement": "unit",
+                        "measurement": "measurement",
+                        "where": "where",
+                        "value_template": "value",
+                        "database": "db2",
+                        "group_function": "fn",
+                        "field": "field",
+                    }
+                ],
+            },
+            {},
+        ),
+        (
+            API_VERSION_2,
+            {
+                "api_version": "2",
+                "ssl": "true",
+                "host": "host",
+                "port": "9000",
+                "path": "path",
+                "token": "token",
+                "organization": "org",
+                "bucket": "bucket",
+                "queries_flux": [
+                    {
+                        "name": "test",
+                        "unit_of_measurement": "unit",
+                        "range_start": "start",
+                        "range_stop": "end",
+                        "group_function": "fn",
+                        "bucket": "bucket2",
+                        "imports": "import",
+                        "query": "query",
+                    }
+                ],
+            },
+            {},
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_full_config(hass, mock_client, config_ext, queries):
+    """Test the full config."""
+    await _setup(hass, config_ext, queries, ["sensor.test"])
+
+
+@pytest.mark.parametrize("config_ext", [(BASE_V1_CONFIG), (BASE_V2_CONFIG)])
+async def test_config_failure(hass, config_ext):
+    """Test an invalid config."""
+    config = {"platform": DOMAIN}
+    config.update(config_ext)
+
+    with pytest.raises(Invalid):
+        PLATFORM_SCHEMA(config)
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock, make_resultset",
+    [
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            _make_v1_resultset,
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            _make_v2_resultset,
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_state_matches_query_result(
+    hass, mock_client, config_ext, queries, set_query_mock, make_resultset
+):
+    """Test state of sensor matches respone from query api."""
+    set_query_mock(mock_client, return_value=make_resultset(42))
+
+    sensors = await _setup(hass, config_ext, queries, ["sensor.test"])
+
+    assert sensors[0].state == "42"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)